### PR TITLE
feat(account_invitation): create access logic

### PR DESCRIPTION
**Pull request by VladislavSokov**

### CODE REVIEW

File: app/views/my_accounts/show.html.erb [hunk 1]
```diff
@@ -2,19 +2,22 @@
 <br>
 Account balance: <%= account.balance %>
 <br>
-<% account.children.each do |child| %>
+<% accounts.each do |account| %>
```
Comment (ka8725) [inline app/views/my_accounts/show.html.erb:5]: I have another idea - what if we just put a section below called **Shared accounts** and list these shared accounts there?

Also, later we could show not accepted invitations there as well (creating a ticket for that)

**Review by andreybakanovsky — APPROVED** [review]

File: app/controllers/my_accounts_controller.rb [hunk 2]
```diff
@@ -8,4 +10,10 @@ def show; end
   helper_method memoize def account
     current_user.account
   end
+
+  helper_method def accounts
+    Account.select('accounts.*, users.email as user_email, users.id as user_id')
+           .invitees(current_user)
+           .joins('INNER JOIN users ON users.account_id = accounts.parent_id')
+  end
```
Comment (ka8725) [inline app/controllers/my_accounts_controller.rb:18]:
```ruby
helper_method memoize def shared_accounts
  Account.invitees(current_user)
end
```

File: app/models/account.rb [hunk 3]
```diff
@@ -15,7 +17,13 @@ class Account < ApplicationRecord
   validates :name, presence: true
 
   scope :visible_for, lambda { |current_user|
-    where(id: [current_user.account_id] + current_user.account.child_ids)
+    where(id: [current_user.account_id] +
+               current_user.account.child_ids +
+               AccountInvitation.accepted.for(current_user).pluck(:account_id))
```
Comment (ka8725) [inline app/models/account.rb:22]:
```suggestion
               invitees(current_user).pluck(:id)
```

File: spec/controllers/my_accounts_spec.rb [hunk 4]
```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
+
 require 'rails_helper'
 
 RSpec.describe MyAccountsController, type: :controller do
```
Comment (ka8725) [inline spec/controllers/my_accounts_spec.rb:5]: add a test to check the shared accounts

**Review by ka8725 — CHANGES_REQUESTED** [review]

File: .rubocop_todo.yml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # This configuration was generated by
 # `rubocop --auto-gen-config --exclude-limit 10000`
-# on 2023-08-22 14:43:11 UTC using RuboCop version 1.56.0.
+# on 2023-08-25 14:11:22 UTC using RuboCop version 1.56.0.
```
[the remark this answers is not in the document]
Comment (VladislavSokov) [reply .rubocop_todo.yml:3]: I fix all linter warnings in every file I work on. And since these warnings are fixed, I remove them from to_do

**Review by ka8725 — APPROVED** [review]

**Pull request merged**